atomate/vasp/firetasks/exchange.py

```diff
@@ -0,0 +1,289 @@
+from fireworks import FiretaskBase, FWAction, explicit_serialize
+
+from monty.serialization import loadfn, dumpfn
+
+from atomate.utils.utils import get_logger, env_chk
+from atomate.vasp.database import VaspCalcDb
+
+from datetime import datetime
+import numpy as np
+
+from pymatgen import Structure
+from pymatgen.command_line.bader_caller import bader_analysis_from_path
+from pymatgen.analysis.magnetism.heisenberg import HeisenbergMapper, HeisenbergModel
+from pymatgen.analysis.magnetism import CollinearMagneticStructureAnalyzer, Ordering
+from pymatgen.command_line.vampire_caller import VampireCaller
+
+__author__ = "Nathan C. Frey"
+__email__ = "ncfrey@lbl.gov"
+
+logger = get_logger(__name__)
+
+
+@explicit_serialize
+class HeisenbergModelMapping(FiretaskBase):
+    """
+    Map structures and energies to a Heisenberg model and compute exchange
+    parameters for a given NN cutoff.
+
+    * heisenberg_settings: 
+        cutoff (float): Starting point for nearest neighbor search.
+        tol (float): Tolerance for equivalent NN bonds.
+
+    Args:
+        structures (list): Magnetic structures.
+        energies (list): Energies / atom (eV).
+
+    Optional parameters:
+        heisenberg_settings (dict): A config dict for Heisenberg model 
+            mapping, detailed above.
+
+    """
+
+    required_params = ["structures", "energies"]
+
+    optional_params = ["heisenberg_settings"]
+
+    def run_task(self, fw_spec):
+
+        structures = self["structures"]
+        energies = self["energies"]
+
+        heisenberg_settings = self.get("heisenberg_settings", {})
+
+        # Total energies
+        energies = [e * len(s) for e, s in zip(energies, structures)]
+
+        # Map system to a Heisenberg Model
+        hmapper = HeisenbergMapper(structures, energies, **heisenberg_settings)
+
+        # Get MSONable Heisenberg Model
+        hmodel = hmapper.get_heisenberg_model()
+        hmodel_dict = hmodel.as_dict()
+
+        # Update FW spec with model
+        name = "heisenberg_model_" + str(hmodel.cutoff)
```

Comment:
This is the line that is causing confusion.

Can you edit this line to be
```python
name = "heisenberg_model_" + str(hmodel.cutoff).replace(".", "_")
```

Reply:
Got it, thanks!